### DISCUSSION TITLE
Is it possible to change the file extension of files created when using the "Create Declaration / Definition" context menu action?

### DISCUSSION THREAD

**Discussion by starball5** [microsoft/vscode-cpptools#10969] [Q&A] [answered]
It seems (or maybe I'm wrong) that in c++ header files, using the "Create Declaration / Definition" context menu action on a declaration will always create definition files with ".cpp" as the file extension.

Is it possible to change that file extension to anything else? (Ex. ".cc" or ".cxx"?)

---

I tried looking through the source code and found the following code for the extension client that handles the server response, but it's not very enlightening into what the server side does.

https://github.com/microsoft/vscode-cpptools/blob/v1.15.4/Extension/src/LanguageServer/client.ts#L3519

I gave a few minutes of `nm -nC`, `strings`, `objdump -SCd` of my installed cpptools-srv binary, and then threw in the towel figuring I was out of my depth with those tools.

Also potentially loosely related: https://github.com/Microsoft/vscode-cpptools/issues/1341.

**Comment by sean-mcmanus** [comment, marked as answer]
It's not currently possible. I've filed an issue at https://github.com/microsoft/vscode-cpptools/issues/10970 to track that.

**Reply by starball5** [reply]
As a follow up, I just encountered https://stackoverflow.com/q/76918979/11107541, where someone asks for changing the default extension used in the save as dialog. Should a feature-request issue ticket be raised here for this as well if this is wanted?

**Reply by sean-mcmanus** [reply]
@starball5 Our extension doesn't control the C++ save as dialog file extension behavior -- you should file an issue with VS Code itself https://github.com/microsoft/vscode/issues
